core/domain/beam_job_domain.py

```diff
@@ -0,0 +1,191 @@
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects related to Apache Beam jobs."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import python_utils
+import utils
+
+(beam_job_models,) = models.Registry.import_models([models.NAMES.beam_job])
+
+
+class BeamJob(python_utils.OBJECT):
+    """Encapsulates the definition of an Apache Beam job.
+
+    Attributes:
+        name: str. The name of the class that implements the job's logic.
+        parameter_names: list(str). The name of the parameters the job accepts
+            as arguments.
+    """
+
+    def __init__(self, job_class):
+        """Initializes a new instance of BeamJob.
+
+        Args:
+            job_class: type(JobBase). The JobBase subclass which implements the
+                job's logic.
+        """
+        self._job_class = job_class
+
+    @property
+    def name(self):
+        """Returns the name of the class that implements the job's logic.
+
+        Returns:
+            str. The name of the job class.
+        """
+        return self._job_class.__name__
+
+    @property
+    def parameter_names(self):
+        """Returns the name of the parameters the job accepts as arguments.
+
+        Returns:
+            list(str). The name of the parameters.
+        """
+        # We don't want to include 'self' in the list; run() being a method
+        # should be considered an implementation detail.
+        return python_utils.get_args_of_function(self._job_class.run)[1:]
+
+    def to_dict(self):
+        """Returns a dict representation of the BeamJob.
+
+        Returns:
+            dict(str: *). The dict has the following structure:
+                name: str. The name of the class that implements the job's
+                    logic.
+                parameter_names: list(str). The name of the parameters the job
+                    accepts as arguments.
+        """
+        return {
+            'name': self.name,
+            'parameter_names': self.parameter_names,
+        }
+
+
+class BeamJobRun(python_utils.OBJECT):
+    """Encapsulates individual executions of an Apache Beam job.
+
+    Attributes:
+        job_id: str. The ID of the job execution.
+        job_name: str. The name of the job class that implements the job's
```

Comment:
job_type would be a better term I think? Usually a name relates to the entity itself but it seems like this is the name of its class, i.e. its type.

Reply:
That seems like a significant difference -- why would we need functions? The description in the docstring itself says "class".

Having said that I think I'm fine with job_name too, since we aren't naming job executions.

Reply:
Ack'd.